faster_whisper/transcribe.py

```diff
@@ -318,8 +318,11 @@ def transcribe(
         self.logger.info(
             "Processing audio with duration %s", format_timestamp(duration)
         )
-
-        if vad_filter:
+        if isinstance(vad_filter, list):
+            speech_chunks = vad_filter
+            audio = collect_chunks(audio, speech_chunks)
+            duration_after_vad = audio.shape[0] / sampling_rate
```

Comment:
These lines are copy-pasted. It means that the code could be factored (while trying to keep the change minimal)
For instance:
```
if vad_filter:
   ... 
   if isinstance(vad_filter, bool):
      speech_chunks = get_speech_timestamps(audio, vad_parameters)
   else:
      speech_chunks = vad_filter # maybe with some checks...
   ... after the same happens ...
```